pymongo/asynchronous/bulk.py

```diff
@@ -214,28 +215,45 @@ def add_delete(
             self.is_retryable = False
         self.ops.append((_DELETE, cmd))
 
-    def gen_ordered(self) -> Iterator[Optional[_Run]]:
+    def gen_ordered(self, requests) -> Iterator[Optional[_Run]]:
         """Generate batches of operations, batched by type of
         operation, in the order **provided**.
         """
         run = None
-        for idx, (op_type, operation) in enumerate(self.ops):
+        for idx, request in enumerate(requests):
```

Comment:
I think the goal of this ticket is to avoid inflating the whole generator upfront and only iterate requests as they are needed at the encoding step. For example:
```python
coll.bulk_write((InsertOne({'x': 'large'*1024*1024}) for _ in range(1_000_000))
```

If we inflate all at once like we do here, then that code will need to allocate all 1 million documents at once.